main.go

```diff
@@ -133,6 +137,46 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getSize(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
+	if err != nil {
+		return 0, fmt.Errorf("error creating the request for %s: %w", u, err)
```

Comment:
It is a good practice to enrich the error message with context-specific information. That makes debugging much easier.

```suggestion
		return 0, fmt.Errorf("Error fetching content size -- creating the request for %s: %w", u, err)
```